### PR TITLE
Use env values to configure DS after checkpoint

### PR DESCRIPTION
**Pull request by tjwatson**

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java [hunk 1]
```diff
@@ -73,7 +77,23 @@ public static void setupDataSourcePropertiesForCheckpoint(LibertyServer serv, Jd
...
+    public static void setupDataSourceEnvForCheckpoint(LibertyServer serv, JdbcDatabaseContainer<?> cont) throws Exception {
+        java.util.Properties envProps = serv.getServerEnv();
+        envProps.put(DB_DRIVER, DatabaseContainerType.valueOf(cont).getDriverName());
+        boolean isDerby = DatabaseContainerType.valueOf(cont) == DatabaseContainerType.Derby || DatabaseContainerType.valueOf(cont) == DatabaseContainerType.DerbyClient;
+        if (!isDerby) {
+            envProps.put(DB_HOST, cont.getHost());
+            envProps.put(DB_PORT, Integer.toString(cont.getFirstMappedPort()));
+            envProps.put(DB_USER, cont.getUsername());
+            envProps.put(DB_PASSWORD, cont.getPassword());
+        }
```
Comment (KyleAure) [inline dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java:92]:
```suggestion

```

We don't need to worry about the connection properties.  That is already taken care of when we `modifyDataSourcePropsForDatabase` is called.

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java [hunk 2]
```diff
@@ -73,7 +77,23 @@ public static void setupDataSourcePropertiesForCheckpoint(LibertyServer serv, Jd
...
+        File serverEnvFile = new File(serv.getFileFromLibertyServerRoot("server.env").getAbsolutePath());
+        try (OutputStream out = new FileOutputStream(serverEnvFile)) {
+            envProps.store(out, "");
+        }
```
Comment (KyleAure) [inline dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java:96]:
```suggestion
        File serverEnvFile;
        if (serv.fileExistsInLibertyServerRoot("server.env")) {
            serverEnvFile = new File(serv.getFileFromLibertyServerRoot("server.env").getAbsolutePath());
        } else {
            serverEnvFile = new File(serv.getInstallRoot() + "/" + "server.env");
            serverEnvFile.createNewFile();
        }

        try (OutputStream out = new FileOutputStream(serverEnvFile)) {
            envProps.store(out, "");
        }
```

I think from earlier testing `serv.getFileFromLibertyServerRoot("server.env")` will throw FileNotFoundException if the file doesn't already exist.
We should handle preemptively handle that.

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java [hunk 3]
```diff
@@ -305,22 +325,33 @@ private static void modifyDataSourcePropsGeneric(Map<String, DataSource> datasou
         serv.updateServerConfiguration(cloneConfig);
     }
 
+    private static final String DB_DRIVER = "DB_DRIVER";
+    private static final String DB_USER = "DB_USER";
+    private static final String DB_PASSWORD = "DB_PASSWORD";
+    private static final String DB_HOST = "DB_HOST";
+    private static final String DB_PORT = "DB_PORT";
```
Comment (KyleAure) [inline dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerUtil.java:332]:
```suggestion
```

**Review by KyleAure — CHANGES_REQUESTED** [review]

**Comment by tjwatson** [comment]
Not planned

**Pull request closed without merging**